letsencrypt/client/tests/acme_test.py

```diff
@@ -64,31 +65,19 @@ def setUp(self):
         self.nonce = '\xec\xd6\xf2oYH\xeb\x13\xd5#q\xe0\xdd\xa2\x92\xa9'
         self.b64nonce = '7Nbyb1lI6xPVI3Hg3aKSqQ'
 
-    def _validate(self, msg):
+    @classmethod
+    def _validate(cls, msg):
         from letsencrypt.client.acme import SCHEMATA
         jsonschema.validate(msg, SCHEMATA[msg['type']])
 
-    def _signature(self, sig):
-        return {
-            'nonce': self.b64nonce,
-            'alg': 'RS256',
-            'jwk': {
-                'kty': 'RSA',
-                'e': 'AQAB',
-                'n': 'rHVztFHtH92ucFJD_N_HW9AsdRsUuHUBBBDlHwNlRd3fp5'
-                     '80rv2-6QWE30cWgdmJS86ObRz6lUTor4R0T-3C5Q',
-            },
-            'sig': sig,
-        }
-
     def test_challenge_request(self):
         from letsencrypt.client.acme import challenge_request
         msg = challenge_request('example.com')
+        self._validate(msg)
         self.assertEqual(msg, {
             'type': 'challengeRequest',
             'identifier': 'example.com',
         })
-        self._validate(msg)
```

Comment:
`self._validate(msg)` has been just moved up in this particular instance, before asserting equality, which does not change the semantics in any way.

If you refer to the removal of `def _signature`, however, then pylint complained about duplicate lines and rightfully so, because signature functionality is already tested in `crypto_utils_test`. Therefore, only part of the created protocol message is tested (using `pop` appropriately).

Does this make sense?